Haha! Max block rlp strikes again!
```
E   execution_testing.specs.helpers.UnexpectedExecutionFailError: Unexpected fail for Block ({'number': 1}):                                                                                       
E      What: Block unexpectedly failed!                                                                                                                                                            
E     Error: "Block RLP size limit exceeded: 9988637 > 8388608" ([BlockException.RLP_BLOCK_LIMIT_EXCEEDED] Block RLP size limit exceeded: 9988637 > 8388608) 
```

We have 1 (2 including the engine format) test that fails when running:
```
uv run fill --fork Osaka -m benchmark --clean -n 12 --evm-bin=../evmone/build/bin/evmone-t8n --gas-benchmark-values=100
```

This is the test: `tests/benchmark/compute/scenario/test_transaction_types.py::test_block_full_data[benchmark-gas-value_100M-fork_Osaka-blockchain_test-zero_byte_True]`

--- a/.github/configs/feature.yaml
+++ b/.github/configs/feature.yaml
@@ -11,11 +11,6 @@ benchmark:
   evm-type: benchmark
   fill-params: --no-html --fork=Osaka --gas-benchmark-values 1,5,10,30,60,100,150 -m benchmark ./tests/benchmark
 
-benchmark_develop:
-  evm-type: benchmark
-  fill-params: --no-html --fork=Osaka --gas-benchmark-values 1,5,10,30,60,100,150 -m "benchmark" ./tests/benchmark
-  feature_only: true
-
 benchmark_fast:
   evm-type: benchmark
   fill-params: --no-html --fork=Osaka --gas-benchmark-values 100 -m "benchmark" ./tests/benchmark

--- a/tests/benchmark/compute/instruction/test_storage.py
+++ b/tests/benchmark/compute/instruction/test_storage.py
@@ -178,7 +178,7 @@ def create_benchmark_executor(
                 key_warm=False,
                 original_value=original,
                 current_value=original,
-                new_value=original + 1,
+                new_value=2**256 - 1,
             )
 
     # [index, num]
@@ -498,14 +498,15 @@ def test_storage_access_warm(
     """Benchmark warm storage slot accesses."""
     blocks = []
 
-    if storage_action == StorageAction.WRITE_SAME_VALUE:
-        execution_code_body = Op.SSTORE(0, Op.DUP1)
-    elif storage_action == StorageAction.WRITE_NEW_VALUE:
-        execution_code_body = Op.SSTORE(0, Op.GAS)
-    elif storage_action == StorageAction.READ:
-        execution_code_body = Op.POP(Op.SLOAD(0))
-    else:
-        execution_code_body = Bytecode()
+    match storage_action:
+        case StorageAction.WRITE_SAME_VALUE:
+            execution_code_body = Op.SSTORE(0, Op.DUP1)
+        case StorageAction.WRITE_NEW_VALUE:
+            execution_code_body = Op.SSTORE(0, Op.GAS)
+        case StorageAction.READ:
+            execution_code_body = Op.POP(Op.SLOAD(0))
+        case _:
+            raise ValueError("Unspecified storage action")
 
     execution_code = Op.SLOAD(0) + While(body=execution_code_body)
     execution_code_address = pre.deploy_contract(code=execution_code)
@@ -571,14 +572,15 @@ def test_storage_access_warm_benchmark(
     Each iteration accesses a different storage slot (incrementing key)
     to ensure warm access costs are measured.
     """
-    if storage_action == StorageAction.WRITE_SAME_VALUE:
-        attack_block = Op.SSTORE(Op.PUSH0, Op.PUSH0)
-    elif storage_action == StorageAction.WRITE_NEW_VALUE:
-        attack_block = Op.SSTORE(Op.PUSH0, Op.GAS)
-    elif storage_action == StorageAction.READ:
-        attack_block = Op.SLOAD(Op.PUSH0)
-    else:
-        attack_block = Bytecode()
+    match storage_action:
+        case StorageAction.WRITE_SAME_VALUE:
+            attack_block = Op.SSTORE(Op.PUSH0, Op.PUSH0)
+        case StorageAction.WRITE_NEW_VALUE:
+            attack_block = Op.SSTORE(Op.PUSH0, Op.GAS)
+        case StorageAction.READ:
+            attack_block = Op.SLOAD(Op.PUSH0)
+        case _:
+            raise ValueError("Unspecified storage action")
 
     benchmark_test(
         target_opcode=Op.SLOAD

--- a/tests/benchmark/compute/scenario/test_transaction_types.py
+++ b/tests/benchmark/compute/scenario/test_transaction_types.py
@@ -190,15 +190,15 @@ def test_block_full_of_ether_transfers(
 
 
 @pytest.fixture
-def total_cost_floor_per_token() -> int:
-    """Total cost floor per token."""
-    return 10
+def total_cost_floor_per_token(fork: Fork) -> int:
+    """Total cost floor per token (EIP-7623)."""
+    return fork.gas_costs().G_TX_DATA_FLOOR_TOKEN_COST
 
 
 @pytest.fixture
-def total_cost_standard_per_token() -> int:
-    """Total cost floor per token."""
-    return 4
+def total_cost_standard_per_token(fork: Fork) -> int:
+    """Standard cost per token (EIP-7623)."""
+    return fork.gas_costs().G_TX_DATA_STANDARD_TOKEN_COST
 
 
 def calldata_generator(
@@ -251,13 +251,37 @@ def test_block_full_data(
     tx_gas_limit: int,
     fork: Fork,
 ) -> None:
-    """Test a block with empty payload."""
+    """Test a block full of calldata, respecting RLP size limits."""
     iteration_count = math.ceil(gas_benchmark_value / tx_gas_limit)
 
-    gas_remaining = gas_benchmark_value
+    # check for EIP-7934 block RLP size limit and cap gas to stay under it
+    block_rlp_limit = fork.block_rlp_size_limit()
+    effective_gas = gas_benchmark_value
+
+    if block_rlp_limit:
+        # Max calldata bytes at 99% of limit (Osaka: 8,388,608 * 0.99 ≈ 8.3 MB)
+        safe_calldata_bytes = int(block_rlp_limit * 0.99)
+
+        # convert to gas: zero bytes = 10 gas/byte, non-zero = 40 gas/byte
+        gas_per_byte = (
+            total_cost_floor_per_token
+            if zero_byte
+            else total_cost_floor_per_token * 4
+        )
+        # For zero bytes: 8.3MB * 10 = 83M gas just for calldata
+        max_calldata_gas = safe_calldata_bytes * gas_per_byte
+        # Add intrinsic cost per tx (Osaka): 83M + 6 txs * 21k ≈ 83.1M total
+        rlp_limited_gas = max_calldata_gas + iteration_count * intrinsic_cost
+
+        # use the min between benchmark target and the RLP limit
+        effective_gas = min(gas_benchmark_value, rlp_limited_gas)
+
+    gas_remaining = effective_gas
     total_gas_used = 0
     txs = []
     for _ in range(iteration_count):
+        if gas_remaining <= intrinsic_cost:
+            break
         gas_available = min(tx_gas_limit, gas_remaining) - intrinsic_cost
         data = calldata_generator(
             gas_available,

